internal/webhook/sparkpod_defaulter.go

```diff
@@ -490,9 +485,40 @@ func addPriorityClassName(pod *corev1.Pod, app *v1beta2.SparkApplication) error
 			pod.Spec.PreemptionPolicy = nil
 		}
 	}
+}
+
+func addDriverExecutorSelfPriorityClassName(pod *corev1.Pod, app *v1beta2.SparkApplication) error {
+	var priorityClassName *string
+	if util.IsDriverPod(pod) {
+		priorityClassName = app.Spec.Driver.PriorityClassName
+	}
+	if util.IsExecutorPod(pod) {
+		priorityClassName = app.Spec.Executor.PriorityClassName
+	}
+
+	podPriorityClassAssign(pod, priorityClassName)
+
 	return nil
 }
 
+func addDriverExecutorUnifyPriorityClassName(pod *corev1.Pod, app *v1beta2.SparkApplication) error {
+	var priorityClassName *string
+	if app.Spec.BatchSchedulerOptions != nil {
+		priorityClassName = app.Spec.BatchSchedulerOptions.PriorityClassName
+	}
+
+	podPriorityClassAssign(pod, priorityClassName)
+
+	return nil
+}
+func addPriorityClassName(pod *corev1.Pod, app *v1beta2.SparkApplication) error {
+	if *app.Spec.BatchSchedulerOptions.PriorityClassSplitting {
+		return addDriverExecutorSelfPriorityClassName(pod, app)
+	} else {
+		return addDriverExecutorUnifyPriorityClassName(pod, app)
+	}
+}
```

Comment:
From my understanding, the `priorityClassName` in `batchSchedulerOptions` is utilized by Volcano's PodGroup, and it is unrelated to the priorityClass of Spark pods. Given this, it seems unnecessary to have the `prioryClassSplitting` option or the `addDriverExecutorUnifyPriorityClassName`  function. WDYT?

Reply:
OK，prioryClassSplitting batchSchedulerOptions may be a bit inappropriate, let me clarify my usage scenario

The current business scenario has not yet used the ability of podGroup. The underlying K8s cluster is divided into multiple resource pools, and the SparkApplication task itself is defined with different priorities at the business level. Moreover, the resources of the underlying K8s cluster are limited. At this time, it is expected that high-priority Executors can run first when resources are insufficient, so the PriorityClass priority preemption scheduling mechanism of K8s itself is used

But because the default PriorityClass of SparkApplication is in the batchSchedulerOptions, it means that the driver and executor under the same SparkApplication will be set to the same priority; but in the actual business scenario, there may be problems, such as we know that the driver is the controller of the executor, once the drvier node is preempted and expelled, the entire SparkApplication will fail, and the expected effect of the business is that the Driver Pod cannot be preempted with high priority, and the Executor can be preempted with high priority, so that different priorities must be set for the Driver and Executor, such as setting the highest priority for the Driver, and the Executor can have a low priority, so that after the low-priority Executor is expelled, it will wait for high priority After the level Executor is executed and there are free resources, the Executors of these previously expelled low-priority tasks will be created by the corresponding drvier pod

If this is the case, how should this priority be defined?

Reply:
@Kevinz857 Perhaps we can do as follows: 
- If `spec.batchSchedulerOptions.priorityClassName` is defined, it will automatically be propagated to both driver and executor pods by default, as this is the current behavior so that we will not make a breaking change.
- Furthermore, if either `spec.driver.priorityClassName` or `spec.executor.priorityClassName` is specifically defined, it should take precedence over `spec.batchSchedulerOptions.priorityClassName`. In this way, we can eliminate the need to introduce `spec.batchSchedulerOptions.prioryClassSplitting` option as it is somewhat confusing. Also, users can define different priority classes for driver and executor pods.

Reply:
@ChenYi015 Very good suggestion, I can modify the code and resubmit MR according to this proposal.

Reply:
@ChenYi015 According to the latest proposal, if  if either spec.driver.priorityClassName or spec.executor.priorityClassName is specifically defined, it should take precedence over spec.batchSchedulerOptions.priorityClassName.

Modified the code according to this proposal, please help review it when you have time, thanks